docs/agents.md

```diff
@@ -413,7 +413,40 @@ _(This example is complete, it can be run "as is" — you'll need to add `asynci
 
 You can retrieve usage statistics (tokens, requests, etc.) at any time from the [`AgentRun`][pydantic_ai.agent.AgentRun] object via `agent_run.usage()`. This method returns a [`RunUsage`][pydantic_ai.usage.RunUsage] object containing the usage data.
 
-Once the run finishes, `agent_run.result` becomes a [`AgentRunResult`][pydantic_ai.agent.AgentRunResult] object containing the final output (and related metadata).
+Once the run finishes, `agent_run.result` becomes an [`AgentRunResult`][pydantic_ai.agent.AgentRunResult] object containing the final output.
+
+#### Run metadata
+
+Metadata configured on an [`Agent`][pydantic_ai.agent.Agent] or passed to a run is available after completion via [`AgentRun.metadata`][pydantic_ai.agent.AgentRun], [`AgentRunResult.metadata`][pydantic_ai.agent.AgentRunResult], or [`StreamedRunResult.metadata`][pydantic_ai.result.StreamedRunResult].
```

Comment:
Let's start by introducing the feature, with an example of when it could be useful, and then explain how to set it and access it.

We should mention that it's available on `RunContext` and is set on the span attributes (which can be one of the examples of when it can be useful)

And we should mention that it can take a static dict or a function taking `RunContext`. And that it's run at the start and the end of the run.